cmake/dependencies.cmake

```diff
@@ -3,14 +3,13 @@ cmake_minimum_required(VERSION 3.2)
 
 include(${PROJECT_SOURCE_DIR}/cmake/scripts.cmake)
 
-# the names of the targets need to differ from the library filenames
-# this is especially an issue with libcairo, where the library is called libcairo
-# therefore, all libs imported this way have been prefixed with lib
-import_pkgconfig_target(TARGET_NAME libglib PKGCONFIG_TARGET glib-2.0>=2.40)
-import_pkgconfig_target(TARGET_NAME libgobject PKGCONFIG_TARGET gobject-2.0>=2.40)
-import_pkgconfig_target(TARGET_NAME libgio PKGCONFIG_TARGET gio-2.0>=2.40)
-import_pkgconfig_target(TARGET_NAME libzlib PKGCONFIG_TARGET zlib)
-import_pkgconfig_target(TARGET_NAME libcairo PKGCONFIG_TARGET cairo)
+find_package(PkgConfig)
+
+pkg_check_modules(glib glib-2.0>=2.40 IMPORTED_TARGET)
```

Comment:
That feature is only available from CMake 3.6 onwards. We had reasons _not_ to use that feature in the past. I don't think we should change this, especially since I wrote my own import script for this that works on all (relevant) versions of CMake.

Reply:
Our build environment has cmake 3.9.2 so there is nothing wrong on using this feature. Also newer version of CMake can be easily used in the old Centos image.

Writing our own code is a regular case of "Reinventing the wheel". And that's the kind of stuff we should avoid.

Reply:
The point is pkg-config + IMPORTED_TARGET it's enough in this case, it's maintained by the CMake team and works good. Why should we have our own copy?

Reply:
We don't have our own copy. All my script does is import a library from a given set of variables. You just pass it the library prefix, and it'll import the target.

Again: it works fine on all versions of CMake I tested it with (virtually anything > 3.0), it's simple code, and it showcases how this can be done. Also, being able to add debug messages there helped resolve loads of linking issues in the past (e.g., when mixing system and bundled libraries). I first started by maintaining all the different systems by hand and wiring everything manually, and my import system reduced the amount of code significantly. Check the history if you want to, but believe me, it's taken me a while to develop this, and again, it's been working fine for many, many months.

I'm 100% fine with you using the imported target for the exported config, but please restore the original behavior here again. It's still running anyway, you're just not using those libraries any more.

Reply:
I see it as an opportunity to reduce our code base which implies reducing the sources of error and the maintenance work. I guess that you can get a similar debug information by running `cmake --trace`.

Reply:
As said, the code won't go away, as it's used to import all kinds of other targets in a standardized way (e.g., ExternalProject, `find_package(...)`), resulting in targets that all work the same. For that reason, I'd rather continue to use the own code.

Reply:
I'm not saying that we should replace all your code, just let's not use this specific part in the public interface.

Reply:
This is not the public interface. This is the internal code base. For the public interface, using this scheme would require shipping our scripts, which isn't really viable, so using `pkg-config` there is fine.

Reply:
When the libappimage target is exported the names of the other targets linked to it in "public" are also used to generate the `libappimageTargets.cmake` file. So the names of such target must be available in the clients build project in order to properly do the linkage.

That's why we should keep the same target names in both sides. So in order to use our implementation we must make it generate names like PkgConfig::glib so it will match with the ones generated by `pkg-config-moule`. Or manually write our own `libappimageTargets.cmake` which means more maintenance work.

Reply:
I don't understand that. Why is that only an issue for packages imported via `PkgConfig`? We import external project libraries, we import stuff via `find_package`, and there, it has never been an issue...

Reply:
Sounds strange. I want to test that myself later to understand this behavior, if you don't mind the additional delay.

Reply:
I have no rush on it, please take your time.